src/common.jl

```diff
@@ -14,3 +14,80 @@ test_dicts(match::Function, a::Dict, b::Dict) =
             end
         end
     end
+
+"""
+$(TYPEDSIGNATURES)
+
+Internal helper function that determines if a reaction has a gene reaction rule
+and that each gene in the rule is contained in the model.
+"""
+function _has_sensible_gpr(model, rid)
+    grrs = reaction_gene_association(model, rid)
+    isnothing(grrs) && return false
+    isempty(grrs) && return false
+    any("" in grr for grr in grrs) && return false
+
+    gids = Set(reduce(vcat, grrs))
+    all(in.(gids, Ref(genes(model))))
+end
+
+"""
+$(TYPEDSIGNATURES)
+
+Determine if a reaction is probably a transport reaction by checking if:
+1. it has sbo annotations corresponding to a transport reaction
+2. the reaction contains metabolites from at least 2 different compartments
+3. if at least 1 metabolite does not undergo a chemical transformation (via
+   formula or annotation checks)
+
+Note, PTS type transport reactions will be missed if they do not have sbo
+annotations. This test may yield false negatives.
+"""
+function _probably_transport_reaction(model, rid, test_annotation)
+    is_transport_reaction(model, rid) && return true
+    length(Set([metabolite_compartment(model, mid) for mid in metabolites(model)])) == 1 && return false
+
+    comp_mid = Dict{String,Set{String}}()
+    for mid in keys(reaction_stoichiometry(model, rid))
+        comp = metabolite_compartment(model, mid)
+        if haskey(comp_mid, comp)
+            push!(comp_mid[comp], mid)
+        else
+            comp_mid[comp] = Set([mid])
+        end
```

Comment:
AFAIK something like this could work:
```suggestion
        push!(get(comp_mid, comp, Set{String}()), mid)
```

Reply:
ah yeah except it needs the magickal `get!` :D

Reply:
this is great! I didn't know about it!